src/main/job-queues/job-queue-base.ts

```diff
@@ -62,8 +64,10 @@ export class JobQueueBase implements JobQueue {
     resumeConsumer(): void
     // eslint-disable-next-line @typescript-eslint/require-await
     async resumeConsumer(): Promise<void> {
```

Comment:
Resume is called on `drain` events emitted by Piscina, which results in multiple calls to sync state. This is okay, as long as dequeue code is idempotent (which it isn't, thus leading to race conditions).

We didn't face this with graphile because the runner object is unchanged over multiple sync state calls.